products/views.py

```diff
@@ -0,0 +1,161 @@
+                    {
+                        "products": ProductSerializer(queryset, many=True).data,
+                        "status_code": status.HTTP_200_OK,
+                    }
+                )
+            return Response(
+                {
+                    "message": "No products found",
+                    "status_code": status.HTTP_400_BAD_REQUEST,
+                }
+            )
+        return Response(
+            {"message": "Bad request", "status_code": status.HTTP_400_BAD_REQUEST}
+        )
+
+    def retrieve(self, request, pk=None):
+        """
+        Retrieve a specific product
+        Args:
+            request(HttpRequest): Value containing request data
+            pk(int): id(primary_key) of Product
+        Returns:
+            (dict): Value containing a specific product
+        """
+        if request.method == "GET":
+            product = get_object_or_404(Product, pk=pk)
+            return Response(
+                {
+                    "product": ProductSerializer(product).data,
+                    "status_code": status.HTTP_200_OK,
+                }
+            )
+        return Response(
+            {"message": "Bad request", "status_code": status.HTTP_400_BAD_REQUEST}
+        )
+
+    def update(self, request, pk=None):
+        """
+        Updates an existing product
+        Args:
+            request(HttpRequest): Value containing request data
+            pk(int): id(primary_key) of Product
+        Returns:
+            (dict): Value containing new/updated product data
+        """
+        product = get_object_or_404(Product, pk=pk)
+        serializer = ProductSerializer(product, data=request.data, partial=True)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(
+                {
+                    "messagae": "Changes updated successfully",
+                    "status_code": status.HTTP_200_OK,
+                    "updated_data": serializer.data,
+                }
+            )
+        return Response(
+            {
+                "message": "Bad Request",
+                "status_code": status.HTTP_400_BAD_REQUEST,
+                "errors": serializer.errors,
+            }
+        )
+
+    def destroy(self, request, pk=None):
+        """
+        Deletes a product from database
+        Args:
+            request(HttpRequest): Value containing Request data
+            pk(int): id(primary key) of product
+        Returns:
+            (dict): Value containing delete operation status
+        """
+        if request.method == "DELETE":
+            get_object_or_404(Product, pk=pk).delete()
+            return Response(
+                {
+                    "message": "Product deleted successfully",
+                    "status_code": status.HTTP_200_OK,
+                }
+            )
+        return Response(
+            {"message": "Bad request", "status_code": status.HTTP_400_BAD_REQUEST}
```

Comment:
Similar to the above comment, do we need to override this method?

Reply:
I tried not overriding it but it gave me the following error
![image](https://user-images.githubusercontent.com/88365652/133744529-b3a46383-8ed7-4b7d-9f6a-f76102a3cc71.png)

Reply:
Even if this error is resolved we will still need to override this method because by default `.destroy()` returns a `No Content` response which does not tell anything. We would still need to override it to return a verbose message. That's what my research told me. Please let me know if that is not the case